dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerImportService.java

```diff
@@ -102,19 +102,28 @@ public TrackerImportReport importTracker( TrackerImportParams params )
             notifier.notify( params.getJobConfiguration(), "(" + params.getUsername() + ") Import:Start" );
         }
 
-        List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
-
-        trackerBundles = preProcessBundle( trackerBundles, importReport );
-
-        TrackerValidationReport validationReport = validateBundle( params, importReport, trackerBundles );
-
-        if ( validationReport.hasErrors() )
+        if ( TrackerImportStrategy.DELETE == params.getImportStrategy() )
         {
-            importReport.setStatus( TrackerStatus.ERROR );
+            List<TrackerBundle> trackerBundles = preheatBundle( params, importReport );
```

Comment:
This declaration is repeated in the if and in the else, put it outside of the block to remove the duplication of code